hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneVolume.java

```diff
@@ -365,7 +365,25 @@ public Iterator<? extends OzoneBucket> listBuckets(String bucketPrefix) {
    */
   public Iterator<? extends OzoneBucket> listBuckets(String bucketPrefix,
       String prevBucket) {
-    return new BucketIterator(bucketPrefix, prevBucket);
+    return listBuckets(bucketPrefix, prevBucket, false);
+  }
+
+  /**
+   * Returns Iterator to iterate over all buckets after prevBucket in the
+   * volume.
+   * If prevBucket is null it iterates from the first bucket in the volume.
+   * The result can be restricted using bucket prefix, will return all
+   * buckets if bucket prefix is null.
+   *
+   * @param bucketPrefix Bucket prefix to match
+   * @param prevBucket Buckets are listed after this bucket
+   * @param isSnapshot Set the flag to list the buckets which have snapshot
+   * @return {@code Iterator<OzoneBucket>}
+   */
+  public Iterator<? extends OzoneBucket> listBuckets(String bucketPrefix,
+                                                     String prevBucket,
+                                                     boolean isSnapshot) {
```

Comment:
nit: I don't think that `isSnapshot` is a good name here. May be `listSnapshot` or `onlySnapshot` would be better in my opinion.